components/support/payload/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde::Serialize;
+use std::{
+    io::{self, Write},
+    num::NonZeroUsize,
+};
+
+/// A quantity of items that can fit into a payload, accounting for
+/// serialization, encryption, and Base64-encoding overhead.
+pub enum Fit {
+    /// All items in the list can fit into the payload.
+    All,
```

Comment:
I'm not really sure about the value of All vs Some tbh - I can't see how real code would actually differ in how they are treated.  None is a bit borderline but can see this being special.

Reply:
(oops, meant to delete this comment, I think what you have is great!)